src/main/java/com/powsybl/balances_adjustment/pevf/PevfExchangesXml.java

```diff
@@ -0,0 +1,323 @@
+        AtomicReference<ZonedDateTime> end = new AtomicReference<>();
+        AtomicReference<Duration> spacing = new AtomicReference<>();
+        LinkedList<Integer> positions = new LinkedList<>();
+        LinkedList<Double> stepValues = new LinkedList<>();
+        Map<String, String> tags = new HashMap<>();
+        AtomicReference<String> code = new AtomicReference<>();
+        AtomicReference<String> text = new AtomicReference<>();
+
+        XmlUtil.readUntilEndElement(TIMESERIES_ELEMENT_NAME, reader, () -> {
+            switch (reader.getLocalName()) {
+                case MRID_ELEMENT_NAME:
+                    timeSeriesMRID.set(reader.getElementText());
+                    break;
+
+                case BUSINESS_TYPE_ELEMENT_NAME:
+                case PRODUCT_ELEMENT_NAME:
+                case IN_DOMAIN_ELEMENT_NAME + "." + MRID_ELEMENT_NAME:
+                case OUT_DOMAIN_ELEMENT_NAME + "." + MRID_ELEMENT_NAME:
+                case CONNECTING_LINE_REGISTERED_RESOURCE_ELEMENT_NAME + "." + MRID_ELEMENT_NAME:
+                case MEASUREMENT_UNIT_ELEMENT_NAME:
+                case CURVE_TYPE_ELEMENT_NAME:
+                    if (reader.getLocalName().equals(IN_DOMAIN_ELEMENT_NAME + "." + MRID_ELEMENT_NAME) ||
+                        reader.getLocalName().equals(OUT_DOMAIN_ELEMENT_NAME + "." + MRID_ELEMENT_NAME)) {
+                        String codingScheme = reader.getAttributeValue(null, CODING_SCHEME_ELEMENT_NAME);
+                        tags.put(reader.getLocalName() + "." + CODING_SCHEME_ELEMENT_NAME, codingScheme);
+                    }
+                    tags.put(reader.getLocalName(), reader.getElementText());
+                    break;
+                case PERIOD_ELEMENT_NAME:
+                    readPeriod(reader, start, end, spacing, positions, stepValues);
+                    break;
+
+                case REASON_ELEMENT_NAME:
+                    readReason(reader, code, text);
+                    break;
+
+                default:
+                    LOGGER.warn(UNEXPECTED_TOKEN, reader.getLocalName());
+            }
+        });
+
+        // Log TimeSeries Reason
+        if (LOGGER.isInfoEnabled()) {
+            LOGGER.info("TimeSeries '{}' [{}, {}] - {} ({}) : {}",  timeSeriesMRID.get(),
+                                                                    start.get(), end.get(),
+                                                                    code.get(),
+                                                                    StandardReasonCodeType.valueOf(code.get()).getDescription(),
+                                                                    text.get());
+        }
+
+        // Create DataChunk
+        DoubleDataChunk dataChunk = null;
+        // Computed number of steps
+        int nbSteps = java.lang.Math.toIntExact((end.get().toInstant().toEpochMilli() - start.get().toInstant().toEpochMilli()) / spacing.get().toMillis());
```

Comment:
remove java.lang prefix
I think the issue with the number of step comes from here.